lib/moonshot/json_stack_template.rb

```diff
@@ -0,0 +1,26 @@
+require 'json'
+require_relative 'stack_template'
+
+module Moonshot
```

Comment:
So, after reviewing all the code in these classes, I suggest:
- Lazy-loading private methods for accessing the template, like:
```ruby
class JsonStackTemplate < StackTemplate
  def body
    template_body.to_json
  end

  private

  def template_body
    @template_body ||= JSON.parse(File.read(@filename))
  end
end
```

- Moving parameters/resource_names into the parent class with the `#fetch` implementations here, since they seem compatible, unless I am missing something.